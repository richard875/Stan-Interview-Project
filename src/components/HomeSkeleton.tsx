import styled from "styled-components";

const HomeSkeleton = () => (
  <Container>
    {[...Array(20)].map((_, i) => (
      <Card data-testid="skeleton-card" key={i} />
    ))}
  </Container>
);

export default HomeSkeleton;

const Container = styled.div`
  gap: 3px;
  display: flex;
  overflow: hidden;
`;

const Card = styled.div`
  margin: 6px;
  width: 200px;
  height: 300px;
  flex-shrink: 0;
  background-color: #606060;
`;
