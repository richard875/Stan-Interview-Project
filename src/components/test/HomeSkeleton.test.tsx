import renderer from "react-test-renderer";
import HomeSkeleton from "../HomeSkeleton";

describe("HomeSkeleton component", () => {
  it("renders correctly", () => {
    const component = renderer.create(<HomeSkeleton />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
