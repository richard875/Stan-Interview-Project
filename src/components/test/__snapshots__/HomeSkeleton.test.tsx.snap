// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeSkeleton component renders correctly 1`] = `
<div
  className="sc-beqWNU leGJVV"
>
  <div
    className="sc-gueXAH hjRfxX"
  />
  <div
    className="sc-gueXAH hjRfxX"
  />
  <div
    className="sc-gueXAH hjRfxX"
  />
  <div
    className="sc-gueXAH hjRfxX"
  />
  <div
    className="sc-gueXAH hjRfxX"
  />
  <div
    className="sc-gueXAH hjRfxX"
  />
  <div
    className="sc-gueXAH hjRfxX"
  />
  <div
    className="sc-gueXAH hjRfxX"
  />
  <div
    className="sc-gueXAH hjRfxX"
  />
  <div
    className="sc-gueXAH hjRfxX"
  />
  <div
    className="sc-gueXAH hjRfxX"
  />
  <div
    className="sc-gueXAH hjRfxX"
  />
  <div
    className="sc-gueXAH hjRfxX"
  />
  <div
    className="sc-gueXAH hjRfxX"
  />
  <div
    className="sc-gueXAH hjRfxX"
  />
  <div
    className="sc-gueXAH hjRfxX"
  />
  <div
    className="sc-gueXAH hjRfxX"
  />
  <div
    className="sc-gueXAH hjRfxX"
  />
  <div
    className="sc-gueXAH hjRfxX"
  />
  <div
    className="sc-gueXAH hjRfxX"
  />
</div>
`;
